sno/working_copy.py

```diff
@@ -749,33 +750,42 @@ def diff_to_tree(self, repo_structure, feature_filter=UNFILTERED):
             )
         return result
 
-    def commit_callback(self, dataset, action, **kwargs):
+    def reset_tracking_table(self, reset_filter=UNFILTERED):
+        reset_filter = reset_filter or UNFILTERED
+
         with self.session() as db:
             dbcur = db.cursor()
+            if reset_filter == UNFILTERED:
+                dbcur.execute(f"DELETE FROM {self.TRACKING_TABLE};")
+                return
 
-            if action in ("I", "U", "D", "META"):
-                pass
-
-            elif action == "INDEX":
-                dbcur.execute(
-                    f"DELETE FROM {self.TRACKING_TABLE} WHERE table_name=?;",
-                    (dataset.name,),
-                )
+            for dataset, pks in reset_filter.items():
+                if pks == UNFILTERED:
+                    dbcur.execute(
+                        f"DELETE FROM {self.TRACKING_TABLE} WHERE table_name=?;",
+                        (gpkg.ident(dataset),),
```

Comment:
this seems wrong... gpkg.ident double-quotes identifiers (`"mytablename"`), but here the table name is a regular parameter, so it will get single-quoted automatically (`'mytablename'`) by the apsw parameter substitution. If it's double quoted it won't ever match anything in this query

Reply:
Fixed